packages/react/src/AddressInput/AddressInput.tsx

```diff
@@ -15,18 +17,22 @@ function setLine(address: Address, index: number, str: string): Address {
   return { ...address, line };
 }
 
-export interface AddressInputProps {
-  name: string;
-  defaultValue?: Address;
-  onChange?: (value: Address) => void;
-}
+type AddressInputProps = ComplexTypeInputProps<Address>;
 
 export function AddressInput(props: AddressInputProps): JSX.Element {
   const [value, setValue] = useState<Address>(props.defaultValue || {});
+  const { getNestedElement } = useContext(BackboneElementContext);
 
   const valueRef = useRef<Address>();
   valueRef.current = value;
 
+  const stateElement = useMemo(() => getNestedElement(props.path, 'state'), [getNestedElement, props.path]);
+  if (stateElement?.binding) {
+    // TODO{mattlong} is it worth the complexity of subbing in an autocomplete input when
```

Comment:
This is a good question.

I think it's a good idea, and something we should consider.  I'm ok to leave that as a future todo though, as this PR is quite big.

I'm quite intrigued by the idea we discussed about `BackboneElementInput` in a horizontal row, which would potentially simplify all of that logic into a single place.  That would also solve this issue as well (i think?)

Reply:
I don't think so?

The type of `address.state` is still string (even though it now has a binding), so `ElementDefinitionTypeInput` would still wind up returning a plain old `TextInput` instead of one of the autocomplete-based inputs that uses the binding.

Or am I not understanding your suggestion correctly?

Reply:
Filed https://github.com/medplum/medplum/issues/3519